astroquery/mast/collections.py

```diff
@@ -0,0 +1,500 @@
+            spectra = [spectra]
+
+        # set up the download directory and paths
+        if not download_dir:
+            download_dir = '.'
+
+        if curl_flag:  # don't want to download the files now, just the curl script
+
+            download_file = "mastDownload_" + time.strftime("%Y%m%d%H%M%S")
+
+            url_list = []
+            path_list = []
+            for spec in spectra:
+                if spec['SpectrumType'] < 2:
+                    url_list.append('https://hla.stsci.edu/cgi-bin/getdata.cgi?config=ops&dataset={0}'
+                                   .format(spec['DatasetName']))
+
+                else:
+                    url_list.append('https://hla.stsci.edu/cgi-bin/ecfproxy?file_id={0}'
+                                   .format(spec['DatasetName']) + '.fits')
+
+                path_list.append(download_file + "/HSC/" + spec['DatasetName'] + '.fits')
+
+            description_list = [""]*len(spectra)
+            producttype_list = ['spectrum']*len(spectra)
+
+            service = "Mast.Bundle.Request"
+            params = {"urlList": ",".join(url_list),
+                      "filename": download_file,
+                      "pathList": ",".join(path_list),
+                      "descriptionList": list(description_list),
+                      "productTypeList": list(producttype_list),
+                      "extension": 'curl'}
+
+            response = self._portal_api_connection.service_request_async(service, params)
+            bundler_response = response[0].json()
+
+            local_path = os.path.join(download_dir, "{}.sh".format(download_file))
+            self._download_file(bundler_response['url'], local_path, head_safe=True, continuation=False)
+
+            status = "COMPLETE"
+            msg = None
+            url = None
+
+            if not os.path.isfile(local_path):
+                status = "ERROR"
+                msg = "Curl could not be downloaded"
+                url = bundler_response['url']
+            else:
+                missing_files = [x for x in bundler_response['statusList'].keys()
+                                if bundler_response['statusList'][x] != 'COMPLETE']
+                if len(missing_files):
+                    msg = "{} files could not be added to the curl script".format(len(missing_files))
+                    url = ",".join(missing_files)
+
+            manifest = Table({'Local Path': [local_path],
+                              'Status': [status],
+                              'Message': [msg],
+                              "URL": [url]})
+
+        else:
+            base_dir = download_dir.rstrip('/') + "/mastDownload/HSC"
+
+            if not os.path.exists(base_dir):
+                os.makedirs(base_dir)
+
+            manifest_array = []
+            for spec in spectra:
+
+                if spec['SpectrumType'] < 2:
+                    data_url = 'https://hla.stsci.edu/cgi-bin/getdata.cgi?config=ops&dataset=' \
```

Comment:
I slightly prefer multiline strings/fstrings (in parens), than line breaks and the usage of `+`. But this is super low level bike shedding, so if you prefer this it certainly can stay as is